MySpeed Install Fails - [ERROR] in line 45: exit code 0: while executing command $STD npm install

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

MySpeed MySpeed LXC

### 📂 What was the exact command used to execute the script?

bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/myspeed.sh)"

### 📝 Provide a clear and concise description of the issue.

Tried to install MySpeed via the script but ran into the error "[ERROR] in line 45: exit code 0: while executing command $STD npm install".

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 🔄 Steps to reproduce the issue.

Use the [script](bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/myspeed.sh)") in your PVE console. Install with defaul settings. Get error.

### ❌ Paste the full error output (if available).

[...]
  ✔️  Installed Node.js
npm error code 1
npm error path /opt/myspeed/node_modules/sqlite3
npm error command failed
npm error command sh -c prebuild-install -r napi || node-gyp rebuild
npm error make: Entering directory '/opt/myspeed/node_modules/sqlite3/build'
npm error   ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlite-autoconf-3440200/sqlite3.c
npm error   TOUCH Release/obj.target/deps/action_before_build.stamp
npm error   CC(target) Release/obj.target/sqlite3/gen/sqlite-autoconf-3440200/sqlite3.o
npm error make: Leaving directory '/opt/myspeed/node_modules/sqlite3/build'
npm error prebuild-install warn This package does not support N-API version undefined
npm error prebuild-install warn install No prebuilt binaries found (target=undefined runtime=napi arch=x64 libc= platform=linux)
npm error gyp info it worked if it ends with ok
npm error gyp info using node-gyp@8.4.1
npm error gyp info using node@22.14.0 | linux | x64
npm error (node:4588) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
npm error (Use `node --trace-deprecation ...` to show where the warning was created)
npm error gyp info find Python using Python version 3.11.2 found at "/usr/bin/python3"
npm error gyp http GET https://nodejs.org/download/release/v22.14.0/node-v22.14.0-headers.tar.gz
npm error gyp http 200 https://nodejs.org/download/release/v22.14.0/node-v22.14.0-headers.tar.gz
npm error gyp http GET https://nodejs.org/download/release/v22.14.0/SHASUMS256.txt
npm error gyp http 200 https://nodejs.org/download/release/v22.14.0/SHASUMS256.txt
npm error gyp info spawn /usr/bin/python3
npm error gyp info spawn args [
npm error gyp info spawn args   '/opt/myspeed/node_modules/node-gyp/gyp/gyp_main.py',
npm error gyp info spawn args   'binding.gyp',
npm error gyp info spawn args   '-f',
npm error gyp info spawn args   'make',
npm error gyp info spawn args   '-I',
npm error gyp info spawn args   '/opt/myspeed/node_modules/sqlite3/build/config.gypi',
npm error gyp info spawn args   '-I',
npm error gyp info spawn args   '/opt/myspeed/node_modules/node-gyp/addon.gypi',
npm error gyp info spawn args   '-I',
npm error gyp info spawn args   '/root/.cache/node-gyp/22.14.0/include/node/common.gypi',
npm error gyp info spawn args   '-Dlibrary=shared_library',
npm error gyp info spawn args   '-Dvisibility=default',
npm error gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/22.14.0',
npm error gyp info spawn args   '-Dnode_gyp_dir=/opt/myspeed/node_modules/node-gyp',
npm error gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/22.14.0/<(target_arch)/node.lib',
npm error gyp info spawn args   '-Dmodule_root_dir=/opt/myspeed/node_modules/sqlite3',
npm error gyp info spawn args   '-Dnode_engine=v8',
npm error gyp info spawn args   '--depth=.',
npm error gyp info spawn args   '--no-parallel',
npm error gyp info spawn args   '--generator-output',
npm error gyp info spawn args   'build',
npm error gyp info spawn args   '-Goutput_dir=.'
npm error gyp info spawn args ]
npm error gyp info spawn make
npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm error make: cc: No such file or directory
npm error make: *** [deps/sqlite3.target.mk:142: Release/obj.target/sqlite3/gen/sqlite-autoconf-3440200/sqlite3.o] Error 127
npm error gyp ERR! build error 
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack     at ChildProcess.onExit (/opt/myspeed/node_modules/node-gyp/lib/build.js:194:23)
npm error gyp ERR! stack     at ChildProcess.emit (node:events:518:28)
npm error gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm error gyp ERR! System Linux 6.8.12-8-pve
npm error gyp ERR! command "/usr/bin/node" "/opt/myspeed/node_modules/.bin/node-gyp" "rebuild"
npm error gyp ERR! cwd /opt/myspeed/node_modules/sqlite3
npm error gyp ERR! node -v v22.14.0
npm error gyp ERR! node-gyp -v v8.4.1
npm error gyp ERR! not ok
npm notice
npm notice New major version of npm available! 10.9.2 -> 11.1.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.1.0
npm notice To update run: npm install -g npm@11.1.0
npm notice
npm error A complete log of this run can be found in: /root/.npm/_logs/2025-02-12T17_28_05_855Z-debug-0.log
 ⠇
[ERROR] in line 45: exit code 0: while executing command $STD npm install
root@pve:~#

### 🖼️ Additional context (optional).

Seems the same issue as here: #2284

--- a/install/myspeed-install.sh
+++ b/install/myspeed-install.sh
@@ -1,10 +1,8 @@
 #!/usr/bin/env bash
 
 # Copyright (c) 2021-2025 tteck
-# Author: tteck
-# Co-Author: MickLesk (Canbiz)
-# License: MIT
-# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+# Author: MickLesk (Canbiz)
+# License: MIT | https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
 # Source: https://github.com/gnmyt/myspeed
 
 source /dev/stdin <<<"$FUNCTIONS_FILE_PATH"
@@ -18,6 +16,9 @@ update_os
 msg_info "Installing Dependencies"
 $STD apt-get install -y \
   curl \
+  build-essential \
+  g++ \
+  make \
   sudo \
   make \
   gpg \

